src/ctnetlink/nlas/flow/protocol_info.rs

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: MIT
+
+use std::convert::TryFrom;
+
+use byteorder::{ByteOrder, NativeEndian};
+use netlink_packet_utils::{
+    nla::{Nla, NlaBuffer, NLA_F_NESTED},
+    DecodeError, Emitable, Parseable,
+};
+
+use crate::{
+    constants::{
+        CTA_PROTOINFO_DCCP, CTA_PROTOINFO_SCTP, CTA_PROTOINFO_TCP,
+        CTA_PROTOINFO_TCP_FLAGS_ORIGINAL, CTA_PROTOINFO_TCP_FLAGS_REPLY,
+        CTA_PROTOINFO_TCP_STATE, CTA_PROTOINFO_TCP_WSCALE_ORIGINAL,
+        CTA_PROTOINFO_TCP_WSCALE_REPLY, CTA_PROTOINFO_UNSPEC,
+    },
+    ctnetlink::nlas::ct_attr::{CtAttr, CtAttrBuilder},
+};
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum ProtocolInfo {
+    Tcp(ProtocolInfoTcp),
+    Dccp(CtAttr),
+    Sctp(CtAttr),
+    Other(CtAttr),
```

Comment:
Any linux kernel code confirmed future data will be always `CtAttr` instead of `DefaultNla`?